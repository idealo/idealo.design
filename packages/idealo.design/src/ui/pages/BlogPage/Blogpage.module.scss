.List {
    display: flex;
    flex-wrap: wrap;
    margin: 0 auto;
}

.Content {
    border: 1px solid black;
    // max-width: 400px;
    max-width: 44%;
    padding: 1rem;
    margin: 0 1rem 1rem 0;
    overflow: hidden;
    max-height: 300px;

    img {
        width: 100px;
        float: left;
        margin-right: 1rem;
    }

    h2 {
        margin: 0 auto 1.5rem auto;

        a {
            &:visited {
                color: black;
            }

            color: black;
            text-decoration: none;
        }
    }
}

.ContentBox {
    border: 1px solid black;
    // padding: 1rem;
    margin: 1rem;
    margin-right: 2.5rem;
    overflow: hidden;

    h2, h4 {
        text-align: center;
        margin-bottom: 0;
    }

    img {
        display: block;
        margin-left: auto;
        margin-right: auto;
        width: 90%;
    }
    .socialMediaIcons img {
        width: 30px;
        height: 30px;
        display: inline;
    }

    .ButtonPrevious {
        margin: 1rem;
        margin-right: 2.5rem;
        float: left;
    }

    .ButtonNext {
        margin: 1rem;
        margin-right: 2.5rem;
        float: right;
    }
}

.HighlightedTeaser {
    @extend .ContentBox;

    background-color: orange;
    margin-left: 0;
    margin-right: 1rem;
    min-height: 15rem;
}

