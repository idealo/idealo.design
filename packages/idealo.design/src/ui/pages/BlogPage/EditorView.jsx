import React from 'react'

import Editor from './Editor'
import requiredAuthentication from "./requiredAuthentication";

function EditorView({}) {
  return  <Editor />
}

// console.log('requiredAuthentication',requiredAuthentication);

export default EditorView;