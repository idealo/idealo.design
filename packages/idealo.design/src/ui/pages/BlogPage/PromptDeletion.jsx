import React from "react";
import ReactModal from "react-modal";
import Button from "@storybook/react/dist/demo/Button";

export default function PromptDeletion(props) {
    return (
        <ReactModal
            isOpen={props.show}
            style={{
                overlay: {
                    backgroundColor: 'rgba( 190, 190, 190, 0.75)'
                },
                content: {
                    top: '40%',
                    left: '25%',
                    right: '25%',
                    bottom: '40%',
                }
            }}
        >
            <header>
                <p>{props.message}</p>
            </header>
            <footer>
                <Button onClick={props.onLeave}>Close</Button>
                <Button onClick={props.onArchive}>Archive</Button>
                <Button onClick={props.onDelete}>Delete</Button>
            </footer>

        </ReactModal>
    );
}
