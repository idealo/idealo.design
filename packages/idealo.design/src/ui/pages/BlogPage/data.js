import fetch from "node-fetch"

// const API_BASE = 'http://localhost:8080';
const API_BASE = '';

export async function fetchList(base_url = API_BASE) {
  const resp = await fetch(`${base_url}/api/blogposts`);
  const data = await resp.json();
  return data;
}

export async function updateSinglePost({ base_url = API_BASE,slug, post }, cb) {
  const body = JSON.stringify(post);

  const resp = await fetch(`${base_url}/api/blogposts`, {
        method: 'PUT',
        headers: { 'Accept': 'application/json', 'Content-Type': 'application/json' },
        body
      }
  ).then(function(response) {
      cb();
    return response.json();
  });

}

export async function fetchSinglePost({ slug }) {
  if (!slug) return null;

  const resp = await fetch(`${API_BASE}/api/blogposts/${slug}`);
  const data = await resp.json();
  return data.pop();
}

export async function fetchPostsByCategorySlug({ categorySlug }) {
  if (!categorySlug) return [];

  const resp = await fetch(`${API_BASE}/api/blogposts?byCategorySlug=${categorySlug}`);
  const data = await resp.json();
  return data;
}

export async function fetchAllCategories() {
  const resp = await fetch(`${API_BASE}/api/categories`);
  const data = await resp.json();
  return data;
}

export async function fetchDistinctCategories(base_url = API_BASE) {
    const resp = await fetch(`${base_url}/api/distinctCategories`);
    const data = await resp.json();
    return data;
}

export async function fetchUserInfo(base_url = API_BASE) {
  const resp = await fetch( `${base_url}/api/me` );
  const data = await resp.json();

  return data;
}

export async function deleteSinglePost(post, base_url = API_BASE) {
    const body = JSON.stringify(post);

    await fetch(`${base_url}/api/blogposts/delete`, {
        method: 'DELETE',
        headers: { 'Accept': 'application/json', 'Content-Type': 'application/json' },
        body
    })
    return post
}

export async function archiveSinglePost(post) {
    const body = JSON.stringify(post);

    await fetch(`${API_BASE}/api/blogposts/archive`,{
        method: 'PUT',
        headers: { 'Accept': 'application/json', 'Content-Type': 'application/json' },
        body
    });
}