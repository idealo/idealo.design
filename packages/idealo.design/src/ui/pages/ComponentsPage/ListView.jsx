import React from "react";
import {withRouter} from "react-router";

class ListView extends React.Component {
    constructor(props) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
        this.state = {
            showMenu: false,
        }
        this.showMenu = this.showMenu.bind(this);
        this.closeMenu = this.closeMenu.bind(this);
    }

    handleClick() {
        console.log('Click happened');
    }

    showMenu(event) {
        event.preventDefault();

        this.setState({showMenu: true}, () => {
            document.addEventListener('click', this.closeMenu);
        });
    }

    closeMenu() {
        if (!this.dropdownMenu.contains(event.target)) {

            this.setState({showMenu: false}, () => {
                document.removeEventListener('click', this.closeMenu);
            });

        }
    }


    render() {
        return (
            <div>
                <div>
                    <button onClick={this.showMenu}>By Technology</button>
                    {
                        this.state.showMenu
                            ? (
                                <div className="menu"
                                     ref={(element) => {
                                         this.dropdownMenu = element;
                                     }}>
                                    <div>
                                        <button> React Stacks</button>
                                    </div>
                                    <div>
                                        <button> Classic Stacks</button>
                                    </div>
                                </div>
                            ) : null
                    }
                </div>

                <div className={element}>
                    <button>1 Element</button>
                </div>

            </div>
    )
        ;
    }
}

export default withRouter(ListView);
