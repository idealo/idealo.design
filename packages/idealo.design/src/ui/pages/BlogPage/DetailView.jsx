import React from 'react'
import ReactModal from 'react-modal'
import draftToHtml from '../../../vendor/draftjs-to-html'
import HtmlToReact from 'html-to-react';
import Prompt from './Editor/Prompt';

import s from './Blogpage.module.scss'
import {fetchSinglePost, fetchUserInfo, deleteSinglePost, archiveSinglePost} from './data'

import {withRouter} from "react-router";

export class DetailView extends React.Component {

    constructor(props) {
        super(props);
        const { history } = props;

        this.state = {
            history: history,
            userInfo :[],
            blogpost: {},
            slug : null,
            isPromptOpen: false,
        }
        this.handlePostEdit = this.handlePostEdit.bind(this);
        this.goBack = this.goBack.bind(this);
        this.handlePopup = this.handlePopup.bind(this);
        this.handleDeletion = this.handleDeletion.bind(this);
        this.handleArchive = this.handleArchive.bind(this);
        this.onModalLeave = this.onModalLeave.bind(this);
    }

    async componentDidMount() {
        const slug = this.props.match.params.slug
        if (slug) {
            this.setState({
                blogpost : await fetchSinglePost({ slug }),
                slug : slug
            })
        }
        this.setState( {
            userInfo : await fetchUserInfo()
        })
        ReactModal.setAppElement('body');
    }

    async componentDidUpdate(prevProps) {
        if (prevProps.match.params.slug !== this.props.match.params.slug) {
            this.setState({
                blogpost : await fetchSinglePost({ slug: this.props.match.params.slug }),
                slug : this.props.match.params.slug
            })
        }
    }

    toDateFormat_de(inp)
    {
        let date = inp ? new Date(inp) : new Date()

        const year = date.getFullYear();
        const month = date.getMonth()+1;
        const day = date.getDate();
        const hour = date.getHours()+2;
        const minute = String(date.getMinutes()).padStart(2,'0');

        return `${day}.${month}.${year} um ${hour}:${minute} Uhr`;
    }

    handlePostEdit()
    {
        this.state.history.push({
            pathname: `/blog/${this.state.slug}/edit`,
            search: `?slug=${this.state.slug}`
        });
    }

    scrollToTop() {
        document.body.scrollTop = 0;
    }

    goBack() {
        this.state.history.push({
            pathname: `/blog/`,
        });
    }

    handleDeletion()
    {
        deleteSinglePost(this.state.blogpost).then(r => this.state.history.push('/blog'))
    }

    handleArchive() {
        archiveSinglePost(this.state.blogpost,
            ()=>{
                this.state.history.push('/blog')
            })
    }

    onModalLeave() {
        this.setState({ isPromptOpen: false});
        this.state.history.push(`/blog/${this.state.slug}`)
    }

    handlePopup()
    {
        this.setState({ isPromptOpen: true});
    }

    render() {
        let facebookLink
        let instagramLink
        let twitterLink
        let emailLink
        let githubLink
        if(this.state.blogpost.facebook !== null){
            facebookLink = <a href={this.state.blogpost.facebook}><img alt="" src="https://img.icons8.com/dusk/64/000000/facebook.png"/></a>
        }
        if(this.state.blogpost.instagram !== null){
            instagramLink = <a href={this.state.blogpost.instagram}><img alt="" src="https://img.icons8.com/doodle/48/000000/instagram-new.png"/></a>
        }
        if(this.state.blogpost.twitter !== null){
            twitterLink = <a href={this.state.blogpost.twitter}><img alt="" src="https://img.icons8.com/doodle/48/000000/twitter--v1.png"/></a>
        }

        if(this.state.blogpost.email !== null){
            emailLink = <a href={'mailto:'+this.state.blogpost.email}><img alt="" src="https://img.icons8.com/doodle/48/000000/email-sign.png"/></a>
        }

        if(this.state.blogpost.github !== null){
            githubLink = <a href={this.state.blogpost.github}><img alt="" src="https://maxcdn.icons8.com/Share/icon/Logos/github_filled1600.png"/></a>
        }

        const htmlBlogContent = draftToHtml(this.state.blogpost.blogpostcontent);
        const HtmlToReactParser = HtmlToReact.Parser;
        const htmlToReactParser = new HtmlToReactParser();
        const reactElement = htmlToReactParser.parse(htmlBlogContent);

        const datetime = this.toDateFormat_de(this.state.blogpost.date);
    return (
    <div className={s.ContentBox}>
      <div className={s.Menu}>
        <button onClick={this.goBack}>Go Back</button>
              {this.state.userInfo.status === 'LOGGED_IN'
                  ? <button onClick={this.handlePopup} title='deleteButton'>Delete</button> : <div> </div>}
              {this.state.userInfo.status === 'LOGGED_IN'
                  ? <button onClick={this.handlePostEdit} title='editButton'>Edit</button> : <div> </div>}
      </div>

      <div className={s.ContentDetailView}>
          <div className={s.SocialMediaIcons}>
              {instagramLink}
              {twitterLink}
              {facebookLink}
              {emailLink}
              {githubLink}
          </div>
          <h2 className={s.blogpostTitle}>{this.state.blogpost.title}</h2>
          <div className={s.Autor}>
              {this.state.blogpost.autor}
          </div>
          <h5 className={s.blogpostDate}>{datetime}</h5>
          {reactElement}
          <img aria-label='blogpostImage' alt="" src={this.state.blogpost.image} />
      </div>

      <div className={s.ButtonNavigation}>
          {this.state.blogpost.previouspost && (<a href={`/blog/${this.state.blogpost.previouspost}`} onClick={this.scrollToTop} className={s.ButtonPrevious}>
              <span>Previous</span>
          </a>)}
          {this.state.blogpost.nextpost && (<a href={`/blog/${this.state.blogpost.nextpost}`} onClick={this.scrollToTop} className={s.ButtonNext}>
              <span>Next</span>
          </a>)}
      </div>

      <Prompt
          show = {this.state.isPromptOpen}
          onDelete = {this.handleDeletion}
          onArchive = {this.handleArchive}
          onCancel = {this.onModalLeave}
          message = 'Do you want to delete or archive that post?'
      />
    </div>
    );
    }
}

export default withRouter(DetailView);

