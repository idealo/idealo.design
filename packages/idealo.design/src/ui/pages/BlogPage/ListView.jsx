import React from 'react';

import withStyles from 'isomorphic-style-loader/withStyles'

import {fetchList, fetchPostsByCategorySlug, fetchUserInfo} from './data';
import { Link, useParams, useHistory } from "react-router-dom";

import s from './Blogpage.module.scss';
import draftToHtml from '../../../vendor/draftjs-to-html';
import { htmlToText } from 'html-to-text';
import { withRouter } from "react-router";

export class ListView extends React.Component{

    constructor(props) {
        super(props);
        this.state = {
            userInfo : {},
            list: []
        }
        this.handleNewPost = this.handleNewPost.bind(this)
        this.componentDidMount = this.componentDidMount.bind(this)
        this.getReactElement = this.getReactElement.bind(this)
    }

    async componentDidMount() {
        await this.updateListView();
    }

    async componentDidUpdate(prevProps) {
        if (prevProps.match.params.slug !== this.props.match.params.slug) {
            await this.updateListView();
        }
    }

    async updateListView(){
        try{
            const slug = this.props.match.params.slug
            if(slug){
                this.setState({
                    list : await fetchPostsByCategorySlug({categorySlug: slug})
                })
            }else {
                this.setState({
                    list: await fetchList()
                })
            }
            this.setState({
                userInfo: await fetchUserInfo()
            })
        }catch(error){

        }
    }

    handleNewPost(){
        this.props.history.push("/blog/new-post");

    }

    getReactElement(a){
        const htmlBlogContent = draftToHtml(a);
        const plainText = htmlBlogContent.replace(/<[^>]*>/g, '');
        return htmlToText(plainText);
    }

    render(
    ) {
        return (
            <div>
                {this.state.userInfo.status === 'LOGGED_IN'
                    ? <button style={{
                        float: 'right',
                        marginBottom: '1rem',
                        marginRight: '1rem',
                        backgroundColor: '#395F86',
                        border: 'none',
                        color: 'white',
                        padding: '10px 30px',
                        textAlign: 'center',
                        fontSize: '14px',
                    }} onClick={this.handleNewPost}>New Post</button>
                    : <div> </div>}

                <div style={{ clear: 'both' }}/>


                <div className={s.List}>
                    {this.state.list.map((blogpost) => (
                        <div key={blogpost.id} className={s.Content}>
                            <h2 className={s.blogpostTitle}>
                                <a href={`/blog/${blogpost.slug}`}>{blogpost.title}</a>
                            </h2>
                            <div><img alt="" src={blogpost.image}/>
                                {this.getReactElement(blogpost.blogpostcontent)}
                            </div>
                        </div>
                    ))}
                </div>
            </div>
        )
    }
}
export default withRouter(ListView);