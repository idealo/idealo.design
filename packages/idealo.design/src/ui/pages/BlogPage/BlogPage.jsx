import React, {useEffect, useState} from 'react'

import {
  // BrowserRouter as Router,
  StaticRouter as Router,
  Switch,
  Route, Redirect,
} from "react-router-dom";

import PageLayout from 'Components/PageLayout'

import ListView from './ListView'
import DetailView from './DetailView'
import EditorView from './EditorView'
import ProtectedRoute from "./protectedRoute";
import {fetchUserInfo} from "./data";

export default function BlogPage({ match }) {

  const [ userInfo, setUserInfo ] = useState([]);

  useEffect(() => {
    fetchUserInfo().then(setUser);
  });

  const setUser = (user) => {
    setUserInfo(user);
  }

  return (
    <>
      <Switch>
        <Route exact path='/blog'>
          <ListView />
        </Route>
        <Route exact path='/blog/categories/:slug'>
          <ListView />
        </Route>
        <Route exact path="/blog/new-post">
          {userInfo.status === 'LOGGED_IN' ?
              <EditorView/>
          :
              <Redirect to='/blog'/>
          }
        </Route>
        <Route path='/blog/:slug/edit'>
          <EditorView />
        </Route>
        <Route path='/blog/:slug'>
          <DetailView />
        </Route>
      </Switch>
    </>
  )
}
