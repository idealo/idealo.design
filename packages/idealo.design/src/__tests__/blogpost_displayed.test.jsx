import React from 'react';
import { create } from "react-test-renderer"
import { ListView } from "../ui/pages/BlogPage/ListView";

//A mock-up blogpost gets rendered and displayed correctly in the DetailView

//yet to be implemented

//A mock-up blogpost gets rendered and displayed correctly in the ListView
it('mock-up blogposts gets rendered and displayed correcly in the listview', done => {
    try {
        const listView = create(<ListView />);
        const listViewInstance = listView.getInstance();
        expect(listViewInstance.state.userInfo.status).toString().match('NOT_LOGGED_IN');
        listViewInstance.setState({list: [{
                "id":1111,
                "title":"A mockup blogpost",
                "nextpost":"docker",
                "previouspost":"mein-erstes-mal-mit-react",
                "categorydisplayvalue":"Docker",
                "categoryslug":"docker",
                "slug":"Einstieg-in-die-Welt-der-Datenbanken",
                "date":"2021-01-20T13:46:44.351Z",
                "image":"https://s12.directupload.net/images/210212/bd5j6kn8.jpg",
                "autor":"Mock-up Post Author",
                "email":"mock-up-posts@gmail.com",
                "instagram":null,
                "twitter":null,
                "github":null,
                "facebook":null,
                "blogpostcontent":{
                    "blocks":[{
                        "key":"3aovx",
                        "data":{},
                        "text":"Just some simple mockup text!",
                        "type":"unstyled",
                        "depth":0,
                        "entityRanges":[],
                        "inlineStyleRanges":[]
                    }],
                    "entityMap":{}
                },
                "isarchived":0
        }]
        });
        expect(listView.toJSON()).toMatchSnapshot();
        done();
    }catch (error) {
        done(error)
    }
});