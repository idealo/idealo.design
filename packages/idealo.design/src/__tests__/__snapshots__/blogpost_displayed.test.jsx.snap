// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mock-up blogposts gets rendered and displayed correcly in the listview 1`] = `
<div>
  <div>
     
  </div>
  <div
    style={
      Object {
        "clear": "both",
      }
    }
  />
  <div
    className="List"
  >
    <div
      className="Content"
    >
      <h2
        className="blogpostTitle"
      >
        <a
          href="/blog/Einstieg-in-die-Welt-der-Datenbanken"
        >
          A mockup blogpost
        </a>
      </h2>
      <div>
        <img
          alt=""
          src="https://s12.directupload.net/images/210212/bd5j6kn8.jpg"
        />
        Just some simple mockup text!
      </div>
    </div>
  </div>
</div>
`;
